In [1]:
import pandas as pd

### 식당데이터

#### Question 20

데이터를 로드하라.

In [2]:
data = pd.read_csv('https://raw.githubusercontent.com/Datamanim/pandas/main/chipo.csv')
df = pd.DataFrame(data)
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


#### Question 21

quantity컬럼 값이 3인 데이터를 추출하여 첫 5행을 출력하라

In [4]:
df[df['quantity'] == 3].head()

# 정답:
Ans = df.loc[df['quantity']==3].head()

#### Question 22

quantity컬럼 값이 3인 데이터를 추출하여 index를 0부터 정렬하고 첫 5행을 출력하라

In [5]:
df[df['quantity'] == 3].head().reset_index()

# 정답
Ans = df.loc[df['quantity']==3].head().reset_index(drop=True)

#### Question 23

quantity , item_price 두개의 컬럼으로 구성된 새로운 데이터 프레임을 정의하라

In [6]:
new_df = pd.merge(df['quantity'],df['item_price'], left_index=True, right_index=True)
new_df

quantity item_price
0            1     $2.39 
1            1     $3.39 
2            1     $3.39 
3            1     $2.39 
4            2    $16.98 
...        ...        ...
4617         1    $11.75 
4618         1    $11.75 
4619         1    $11.25 
4620         1     $8.75 
4621         1     $8.75 

[4622 rows x 2 columns]

#### Question 24 (V)

item_price 컬럼의 달러표시 문자를 제거하고 float 타입으로 저장하여 new_price 컬럼에 저장하라

In [7]:
df['new_price'] = df['item_price'].str[1:].astype('float')

In [8]:
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  new_price  
0                                                   NaN     $2.39        2.39  
1                                          [Clementine]     $3.39        3.39  
2                                               [Apple]     $3.39        3.39  
3                                                   NaN     $2.39        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98       16.98  
...                                                 ...        ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75        8.75  

[4622 rows x 6 columns]

#### Question 25

new_price 컬럼이 5이하의 값을 가지는 데이터프레임을 추출하고, 전체 갯수를 구하여라

In [9]:
df[df['new_price'] < 6]

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
6            3         1                          Side of Chips   
...        ...       ...                                    ...   
4606      1828         1                      Canned Soft Drink   
4609      1829         1                      Canned Soft Drink   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4616      1832         1                    Chips and Guacamole   

     choice_description item_price  new_price  
0                   NaN     $2.39        2.39  
1          [Clementine]     $3.39        3.39  
2               [Apple]     $3.39        3.39  
3                   NaN     $2.39        2.39  
6                   NaN     $1.69        1.69  
...                 ...        ...        ...  
4606             [Coke]     $1.25        1.25  
4609           [Sprite]     $1.25        1.25  
4613                NaN     $2.15        2.15  
4614                NaN     $1.50        1.50  
4616                NaN     $4.45        4.45  

[1661 rows x 6 columns]

In [10]:
print(len(df[df['new_price'] < 6]))

1661


#### Question 26

item_name명이 Chicken Salad Bowl 인 데이터 프레임을 추출하라고 index 값을 초기화 하여라

In [11]:
df[df['item_name'] == 'Chicken Salad Bowl'].reset_index()

index  order_id  quantity           item_name  \
0       44        20         1  Chicken Salad Bowl   
1      135        60         2  Chicken Salad Bowl   
2      213        94         2  Chicken Salad Bowl   
3      256       111         1  Chicken Salad Bowl   
4      315       137         2  Chicken Salad Bowl   
..     ...       ...       ...                 ...   
105   4561      1813         2  Chicken Salad Bowl   
106   4580      1822         1  Chicken Salad Bowl   
107   4619      1834         1  Chicken Salad Bowl   
108   4620      1834         1  Chicken Salad Bowl   
109   4621      1834         1  Chicken Salad Bowl   

                                    choice_description item_price  new_price  
0    [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75        8.75  
1    [Tomatillo Green Chili Salsa, [Sour Cream, Che...    $22.50       22.50  
2    [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $22.50       22.50  
3    [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...     $8.75        8.75  
4              [Fresh Tomato Salsa, Fajita Vegetables]    $17.50       17.50  
..                                                 ...        ...        ...  
105  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $17.50       17.50  
106  [Fresh Tomato Salsa, [Black Beans, Cheese, Gua...    $11.25       11.25  
107  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25       11.25  
108  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75        8.75  
109  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75        8.75  

[110 rows x 7 columns]

#### Question 27

new_price값이 9 이하이고 item_name 값이 Chicken Salad Bowl 인 데이터 프레임을 추출하라

In [12]:
df[(df['item_name'] == 'Chicken Salad Bowl') & (df['new_price'] < 10)].head()

# 정답
df.loc[(df.item_name == 'Chicken Salad Bowl') & (df.new_price < 10)].head()

order_id  quantity           item_name  \
44         20         1  Chicken Salad Bowl   
256       111         1  Chicken Salad Bowl   
526       220         1  Chicken Salad Bowl   
528       221         1  Chicken Salad Bowl   
529       221         1  Chicken Salad Bowl   

                                    choice_description item_price  new_price  
44   [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75        8.75  
256  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...     $8.75        8.75  
526  [Roasted Chili Corn Salsa, [Black Beans, Sour ...     $8.75        8.75  
528  [Tomatillo Green Chili Salsa, [Fajita Vegetabl...     $8.75        8.75  
529  [Tomatillo Green Chili Salsa, [Fajita Vegetabl...     $8.75        8.75

#### Question 28

df의 new_price 컬럼 값에 따라 오름차순으로 정리하고 index를 초기화 하여라

In [13]:
df[['new_price']].sort_index(axis=1)

new_price
0          2.39
1          3.39
2          3.39
3          2.39
4         16.98
...         ...
4617      11.75
4618      11.75
4619      11.25
4620       8.75
4621       8.75

[4622 rows x 1 columns]

#### Question 29

df의 item_name 컬럼 값중 Chips 포함하는 경우의 데이터를 출력하라

In [14]:
df[df.item_name.str.contains('Chips')]

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
6            3         1                          Side of Chips   
10           5         1                    Chips and Guacamole   
14           7         1                    Chips and Guacamole   
...        ...       ...                                    ...   
4596      1826         1                    Chips and Guacamole   
4600      1827         1                    Chips and Guacamole   
4605      1828         1                    Chips and Guacamole   
4613      1831         1                                  Chips   
4616      1832         1                    Chips and Guacamole   

     choice_description item_price  new_price  
0                   NaN     $2.39        2.39  
3                   NaN     $2.39        2.39  
6                   NaN     $1.69        1.69  
10                  NaN     $4.45        4.45  
14                  NaN     $4.45        4.45  
...                 ...        ...        ...  
4596                NaN     $4.45        4.45  
4600                NaN     $4.45        4.45  
4605                NaN     $4.45        4.45  
4613                NaN     $2.15        2.15  
4616                NaN     $4.45        4.45  

[1084 rows x 6 columns]

In [15]:
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  new_price  
0                                                   NaN     $2.39        2.39  
1                                          [Clementine]     $3.39        3.39  
2                                               [Apple]     $3.39        3.39  
3                                                   NaN     $2.39        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98       16.98  
...                                                 ...        ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75        8.75  

[4622 rows x 6 columns]

In [16]:
df.columns[1]
df.columns[3]
df.columns[5]

'new_price'

#### Question 30

df의 짝수번째 컬럼만을 포함하는 데이터프레임을 출력하라

In [17]:
for i in range(3):
    print(df.columns[(2*i)+ 1])
    

quantity
choice_description
new_price


#### Question 31

df의 new_price 컬럼 값에 따라 내림차순으로 정리하고 index를 초기화 하여라

In [18]:
df.sort_values(by='new_price', ascending=False)

order_id  quantity                     item_name  \
3598      1443        15  Chips and Fresh Tomato Salsa   
3480      1398         3                 Carnitas Bowl   
1254       511         4               Chicken Burrito   
3602      1443         4               Chicken Burrito   
3601      1443         3                Veggie Burrito   
...        ...       ...                           ...   
3936      1578         1                   Canned Soda   
2922      1162         1                 Bottled Water   
1396       567         1                   Canned Soda   
2562      1014         1                   Canned Soda   
1457       591         1                   Canned Soda   

                                     choice_description item_price  new_price  
3598                                                NaN    $44.25       44.25  
3480  [Roasted Chili Corn Salsa, [Fajita Vegetables,...    $35.25       35.25  
1254  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...    $35.00       35.00  
3602  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...    $35.00       35.00  
3601  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...    $33.75       33.75  
...                                                 ...        ...        ...  
3936                                  [Diet Dr. Pepper]     $1.09        1.09  
2922                                                NaN     $1.09        1.09  
1396                                        [Coca Cola]     $1.09        1.09  
2562                                        [Coca Cola]     $1.09        1.09  
1457                                           [Sprite]     $1.09        1.09  

[4622 rows x 6 columns]

#### Question 32

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 인덱싱하라

In [19]:
df[(df['item_name'] == 'Steak Salad') | (df['item_name'] == 'Bowl')]

# 답
df.loc[(df.item_name =='Steak Salad') | (df.item_name =='Bowl')]

order_id  quantity    item_name  \
445        193         3         Bowl   
664        276         1  Steak Salad   
673        279         1         Bowl   
752        311         1  Steak Salad   
893        369         1  Steak Salad   
3502      1406         1  Steak Salad   

                                     choice_description item_price  new_price  
445   [Braised Carnitas, Pinto Beans, [Sour Cream, C...    $22.20       22.20  
664   [Tomatillo-Red Chili Salsa (Hot), [Black Beans...     $8.99        8.99  
673   [Adobo-Marinated and Grilled Steak, [Sour Crea...     $7.40        7.40  
752   [Tomatillo-Red Chili Salsa (Hot), [Black Beans...     $8.99        8.99  
893   [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...     $8.99        8.99  
3502                        [[Lettuce, Fajita Veggies]]     $8.69        8.69

#### Question 33 (V)

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 데이터 프레임화 한 후, item_name를 기준으로 중복행이 있으면 제거하되 첫번째 케이스만 남겨라

In [20]:
Ans = df.loc[(df.item_name =='Steak Salad') | (df.item_name =='Bowl')]
Ans = Ans.drop_duplicates('item_name')
Ans

order_id  quantity    item_name  \
445       193         3         Bowl   
664       276         1  Steak Salad   

                                    choice_description item_price  new_price  
445  [Braised Carnitas, Pinto Beans, [Sour Cream, C...    $22.20       22.20  
664  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...     $8.99        8.99

#### Question 34 (V)

df의 item_name 컬럼 값이 Steak Salad 또는 Bowl 인 데이터를 데이터 프레임화 한 후, item_name를 기준으로 중복행이 있으면 제거하되 마지막 케이스만 남겨라

In [21]:
df[(df['item_name'] == 'Steak Salad') | (df['item_name'] == 'Bowl')].drop_duplicates('item_name', keep='last')

order_id  quantity    item_name  \
673        279         1         Bowl   
3502      1406         1  Steak Salad   

                                     choice_description item_price  new_price  
673   [Adobo-Marinated and Grilled Steak, [Sour Crea...     $7.40        7.40  
3502                        [[Lettuce, Fajita Veggies]]     $8.69        8.69

#### Question 35

df의 데이터 중 new_price값이 new_price값의 평균값 이상을 가지는 데이터들을 인덱싱하라

In [22]:
df[df['new_price'] >= df['new_price'].mean()]

order_id  quantity           item_name  \
4            2         2        Chicken Bowl   
5            3         1        Chicken Bowl   
7            4         1       Steak Burrito   
8            4         1    Steak Soft Tacos   
9            5         1       Steak Burrito   
...        ...       ...                 ...   
4617      1833         1       Steak Burrito   
4618      1833         1       Steak Burrito   
4619      1834         1  Chicken Salad Bowl   
4620      1834         1  Chicken Salad Bowl   
4621      1834         1  Chicken Salad Bowl   

                                     choice_description item_price  new_price  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98       16.98  
5     [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98       10.98  
7     [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75       11.75  
8     [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25        9.25  
9     [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25        9.25  
...                                                 ...        ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75        8.75  

[2890 rows x 6 columns]

#### Question 36

df의 데이터 중 item_name의 값이 Izze 데이터를 Fizzy Lizzy로 수정하라

In [40]:
df['item_name'][df.item_name == 'Izze'] = 'Fizzy Lizzy'

C:\Users\hanjieun_2\AppData\Local\Temp\ipykernel_27356\2262582103.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['item_name'][df.item_name == 'Izze'] = 'Fizzy Lizzy'


In [41]:
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                            Fizzy Lizzy   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  new_price  
0                                                   NaN     $2.39        2.39  
1                                          [Clementine]     $3.39        3.39  
2                                               [Apple]     $3.39        3.39  
3                                                   NaN     $2.39        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98       16.98  
...                                                 ...        ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75        8.75  

[4622 rows x 6 columns]

#### Question 37

df의 데이터 중 choice_description 값이 NaN 인 데이터의 갯수를 구하여라

In [42]:
df['choice_description'].isna().sum()

1246

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
6            3         1                          Side of Chips   
10           5         1                    Chips and Guacamole   
14           7         1                    Chips and Guacamole   
...        ...       ...                                    ...   
4600      1827         1                    Chips and Guacamole   
4605      1828         1                    Chips and Guacamole   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4616      1832         1                    Chips and Guacamole   

     choice_description item_price  new_price  
0                   NaN     $2.39        2.39  
3                   NaN     $2.39        2.39  
6                   NaN     $1.69        1.69  
10                  NaN     $4.45        4.45  
14                  NaN     $4.45        4.45  
...                 ...        ...        ...  
4600                NaN     $4.45        4.45  
4605                NaN     $4.45        4.45  
4613                NaN     $2.15        2.15  
4614                NaN     $1.50        1.50  
4616                NaN     $4.45        4.45  

[1246 rows x 6 columns]

#### Question 38

df의 데이터 중 choice_description 값이 NaN 인 데이터를 NoData 값으로 대체하라(loc 이용)

In [ ]:
df[df['choice_description'].isnull() == True]

In [46]:
for i in range(len(df.index))
    if df.loc[i, 'choice_description'] == 

KeyError: 'choice_description'

#### Question 39

df의 데이터 중 choice_description 값에 Black이 들어가는 경우를 인덱싱하라

#### Question 40

df의 데이터 중 choice_description 값에 Vegetables 들어가지 않는 경우의 갯수를 출력하라

#### Question 41

df의 데이터 중 item_name 값이 N으로 시작하는 데이터를 모두 추출하라

#### Question 42

df의 데이터 중 item_name 값의 단어갯수가 15개 이상인 데이터를 인덱싱하라

#### Question 43

df의 데이터 중 new_price값이 lst에 해당하는 경우의 데이터 프레임을 구하고 그 갯수를 출력하라 lst =[1.69, 2.39, 3.39, 4.45, 9.25, 10.98, 11.75, 16.98]

